### PR TITLE
Fix broken support if `redis.password` not explicitly set

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
Fix broken support if `redis.password` not explicitly set

### CODE REVIEW

File: stable/distribution/templates/distribution-statefulset.yaml [hunk 1]
```diff
@@ -186,6 +193,13 @@ spec:
           {{- end }}
             /opt/jfrog/distribution/app/bin/wrapper.sh;
         env:
+        - name: JF_SHARED_REDIS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "distribution.fullname" . }}-redis-secret
```
Comment (rimusz) [inline stable/distribution/templates/distribution-statefulset.yaml:199]: please add an external secret support for Redis password

File: stable/distribution/values.yaml [hunk 2]
```diff
@@ -132,6 +132,7 @@ redis:
     pullPolicy: IfNotPresent
   port: 6379
   password: ""
+  # existingSecret: <name-of-existing-secret>
```
Comment (rimusz) [inline stable/distribution/values.yaml:135]: add in comment an example of the key:
`key: redis-password`

File: stable/distribution/templates/redis-connection-secret.yaml [hunk 3]
```diff
@@ -1,17 +0,0 @@
-apiVersion: v1
```
Comment (danielezer) [inline stable/distribution/templates/redis-connection-secret.yaml:1]: Why has this been deleted? Where would the password be stored in a case where the user didn't provide an existing secret?

File: stable/distribution/README.md [hunk 4]
```diff
@@ -259,6 +259,7 @@ The following table lists the configurable parameters of the distribution chart
 | `postgresql.affinity`                           | PostgreSQL node affinity                                               | `{}`                                                               |
 | `postgresql.tolerations`                        | PostgreSQL node tolerations                                            | `[]`                                                               |
 | `redis.password`                                | Redis password                                                         | ` `                                                                |
+| `redis.existingSecret`                          | Redis existingSecret for password                                      | ` `                                                                |
```
Comment (danielezer) [inline stable/distribution/README.md:262]: I think we should keep the convention we have in other charts, which is `secretName`

File: stable/distribution/templates/redis-connection-secret.yaml (hunk 3 above)
Comment (chukka) [reply stable/distribution/templates/redis-connection-secret.yaml:1]: I moved this logic to `distribution-secrets.yaml` file

File: stable/distribution/README.md (hunk 4 above)
Comment (chukka) [reply stable/distribution/README.md:270]: @danielezer I took the convention as per upstream chart 
https://github.com/bitnami/charts/blob/master/bitnami/redis/values.yaml#L205

Comment (danielezer) [reply stable/distribution/README.md:270]: ok, makes sense

**Pull request merged**